### PR TITLE
Case Search - remove characters from query string by regex

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://docs.google.com/document/d/14C472msXfKntq07SFvVdvVRALso6yUk9Dj_auIDZx_8/edit#heading=h.dn7ah3nwason

Allow users to specify a regular expression of characters to be removed from incoming search query strings.

This is the first half of enabling this workflow: https://docs.google.com/document/d/1SgfxfD5mYRmmHmV4fhqwCC9IhnjrlHHII8Di3hlPT9U/edit#heading=h.cggylme8836o

@benrudolph 🐡 / 🍥

### CODE REVIEW

File: corehq/apps/case_search/models.py [hunk 1]
```diff
@@ -46,6 +45,33 @@ class Meta(object):
         unique_together = ('domain', 'case_type')
 
 
+class RemoveCharacters(models.Model):
```
Comment (proteusvacuum) [inline corehq/apps/case_search/models.py]: This is a terrible name. I've been having "name block" all day and couldn't come up with anything better...

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 2]
```diff
@@ -64,10 +69,139 @@ def test_add_blacklisted_ids(self):
...
+                                        }
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        },
+                    "size": SIZE_LIMIT,
```
Comment (dimagimon) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/ota/views.py [hunk 3]
```diff
@@ -155,12 +128,61 @@ def search(request, domain):
     return HttpResponse(fixtures, content_type="text/xml; charset=utf-8")
 
 
-def add_blacklisted_owner_ids(search_es, blacklisted_owner_ids):
+def _add_include_closed(search_es, criteria):
+    try:
+        include_closed = criteria.pop('include_closed')
```
Comment (benrudolph) [inline corehq/apps/ota/views.py]: nit: could use our `string_to_boolean` function and catch a ValueError if you don't want to check `'True'`

File: corehq/apps/case_search/models.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/case_search/models.py]: yeah that's tough, maybe `Ignore` is more useful than `Remove`? `IgnorePatterns`?

**Review by benrudolph — COMMENTED** [review]
tests are failing, but looks good

**Comment by benrudolph** [comment]
```
======================================================================
FAIL: corehq.apps.ota.tests.test_search_claim_endpoints:CaseClaimEndpointTests.test_search_query_addition
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/ota/tests/test_search_claim_endpoints.py", line 485, in test_search_query_addition
    run_query_mock.assert_called_with("case_search", expected_query, debug_host=None)
  File "/vendor/lib/python2.7/site-packages/mock.py", line 863, in assert_called_with
    raise AssertionError(msg)
AssertionError: Expected call: run_query('case_search', {'query': {'filtered': {'filter': {'and': [{'term': {'domain.exact': 'test-domain'}}, {'term': {'type.exact': 'case'}}, {'term': {'closed': False}}, {'match_all': {}}]}, 'query': {'bool': {'must': [{'nested': {'path': 'case_properties', 'query': {'filtered': {'filter': {'term': {'case_properties.key': u'name'}}, 'query': {'match': {'case_properties.value': {'query': 'wut', 'fuzziness': '0'}}}}}}}, {'bool': {'should': [{'nested': {'path': 'case_properties', 'query': {'filtered': {'filter': {'term': {'case_properties.key': 'is_inactive'}}, 'query': {'match': {'case_properties.value': {'query': 'yes', 'fuzziness': '0'}}}}}}}, {'nested': {'path': 'case_properties', 'query': {'filtered': {'filter': {'term': {'case_properties.key': 'awaiting_claim'}}, 'query': {'match': {'case_properties.value': {'query': 'yes', 'fuzziness': '0'}}}}}}}]}}]}}}}, 'size': 100}, debug_host=None)
Actual call: run_query('case_search', {'query': {'filtered': {'filter': {'and': [{'term': {'domain.exact': u'test-domain'}}, {'term': {'type.exact': u'case'}}, {'term': {'closed': False}}, {'match_all': {}}]}, 'query': {'bool': {'must': [{'nested': {'path': 'case_properties', 'query': {'filtered': {'filter': {'term': {'case_properties.key': u'commcare_custom_search_query'}}, 'query': {'match': {'case_properties.value': {'query': u'1', 'fuzziness': '0'}}}}}}}, {'nested': {'path': 'case_properties', 'query': {'filtered': {'filter': {'term': {'case_properties.key': u'name'}}, 'query': {'match': {'case_properties.value': {'query': u'wut', 'fuzziness': '0'}}}}}}}, {u'bool': {u'should': [{u'nested': {u'path': u'case_properties', u'query': {u'filtered': {u'filter': {u'term': {u'case_properties.key': u'is_inactive'}}, u'query': {u'match': {u'case_properties.value': {u'query': u'yes', u'fuzziness': u'0'}}}}}}}, {u'nested': {u'path': u'case_properties', u'query': {u'filtered': {u'filter': {u'term': {u'case_properties.key': u'awaiting_claim'}}, u'query': {u'match': {u'case_properties.value': {u'query': u'yes', u'fuzziness': u'0'}}}}}}}]}}]}}}}, 'size': 100}, debug_host=None)
```

File: corehq/apps/case_search/models.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/apps/case_search/models.py]: Yes. That is much better.

**Comment by proteusvacuum** [comment]
@benrudolph made https://github.com/dimagi/commcare-hq/pull/16254/commits/b99a57c02da2ff30d33eca3245fa2d3d84065a2c to fix tests and 
https://github.com/dimagi/commcare-hq/pull/16254/commits/d0edab7cb1f6d2034afb8c5cea310e113f4e86ba to rename everything. 
Could you once-over if you think OK?

**Review by benrudolph — APPROVED** [review]
looks good

**Pull request merged**